lib/livebook/hubs/enterprise_client.ex

```diff
@@ -43,7 +43,11 @@ defmodule Livebook.Hubs.EnterpriseClient do
   end
 
   def send_request(pid, %_struct{} = data) do
-    ClientConnection.send_request(GenServer.call(pid, :get_server), data)
+    if GenServer.call(pid, :connected?) do
+      ClientConnection.send_request(GenServer.call(pid, :get_server), data)
+    else
+      {:transport_error, "connection refused"}
+    end
```

Comment:
```suggestion
    with {:ok, server} <- GenServer.call(pid, :fetch_server) do
      ClientConnection.send_request(server, data)
    end
```

And `:fetch_server` will return either `{:ok, server}` or the internal `connection_error` as an error, WDYT?

Reply:
I like this one, I can define the error and get the `connection_error` directly